### DISCUSSION TITLE
why neli is used for linux?

### DISCUSSION THREAD

**Discussion by bigdogs** [EstebanBorai/local-ip-address#65] [Q&A] [answered]
Hi,
Thanks for your crate! 

 I'm not familiar with linux "neli", But "getifaddrs" is available for both linux and macos,  Why not just reuse "getifaddrs" from macos,  What's the advantage of "neli"

**Comment by LeoBorai** [comment, marked as answer]
Hi @bigdogs! Thanks for opening this question!

`neli` is used instead of `getifaddrs` to get an accurate network interface when searching for the host's local IP address.
Netlink provides a flag called `IFA_LOCAL` under its `rta_type`, ([you can find documentation here](https://www.man7.org/linux/man-pages/man7/rtnetlink.7.html#top_of_page)) which I rely on to retrieve the local IP address.

Given than different linux distributions and configurations may assign different names values for their `AF_INET` family network interfaces its very difficult to guarantee an accurate result. Instead by [establishing a Netlink socket connection](https://github.com/EstebanBorai/local-ip-address/blob/66af2a6b993cf850f4475da9074da1bb9fb8f71a/src/linux.rs#L39) and "asking" for an [`IFA_LOCAL`](https://www.man7.org/linux/man-pages/man7/rtnetlink.7.html#top_of_page) network interface I get an accurate result.

**Reply by bigdogs** [reply]
Thank you for your explanation～
